src/index.ts

```diff
@@ -126,29 +134,30 @@ export function targetConstructorToSchema(
  * Return `storage.validationMetadatas` populated with `constraintMetadatas`.
  */
 function getMetadatasFromStorage(
-  storage: cv.MetadataStorage
-): ValidationMetadata[] {
-  const metadatas: ValidationMetadata[] = (storage as any).validationMetadatas
-  return populateMetadatasWithConstraints(storage, metadatas)
+  storage: IStorage
+): Map<any, ValidationMetadata[]> {
+  const metadatas: Map<any, ValidationMetadata[]> = new Map()
+
+  storage.validationMetadatas.forEach((value, target) => {
+    metadatas.set(target, populateMetadatasWithConstraints(storage, value))
+  })
+  return metadatas
 }
 
 function populateMetadatasWithConstraints(
-  storage: cv.MetadataStorage,
+  storage: IStorage,
   metadatas: ValidationMetadata[]
 ): ValidationMetadata[] {
-  const constraints: ConstraintMetadata[] = (storage as any).constraintMetadatas
-
+  const constraints: Map<any, ConstraintMetadata[]> = (storage as any)
+    .constraintMetadatas
   return metadatas.map((meta) => {
     if (meta.constraintCls) {
-      const constraint = constraints.find(
-        (c) => c.target === meta.constraintCls
-      )
-      if (constraint) {
-        return { ...meta, type: constraint.name }
+      const constraint = constraints.get(meta.constraintCls)
```

Comment:
Does using the public API https://github.com/typestack/class-validator/blob/f42197c2c170efd0de322da428fd7ce872c5a07b/src/metadata/MetadataStorage.ts#L154 work here?